### PR TITLE
Support hash routing

### PR DESCRIPTION
**Pull request by raymondjacobson**
### Description
Add env support for ipfs hosting with hash routing. Some squirly stuff around tag search that I had to clean up, but overall not too bad.

### Dragons

Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?
Touches routing

### How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.- 

Click around quite a bit at both:
- npm run start:ipfs-stage
- npm run start:stage

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -115,6 +117,8 @@
     "build:prod-source-maps": "npm run write-sha && env-cmd ./.env/.env.prod env-cmd ./.env/.env.source-maps env-cmd ./.env/.env.git npm run build && rm -rf build-production && mv build build-production",
     "build:mobile-stage": "env-cmd ./.env/.env.mobile npm run build:stage && rm -rf build-mobile-staging && mv build-staging build-mobile-staging",
     "build:mobile-prod": "env-cmd ./.env/.env.mobile npm run build:prod && rm -rf build-mobile-production && mv build-production build-mobile-production",
+    "build:ipfs-stage": "env-cmd ./.env/.env.ipfs.stage npm run build:stage && rm -rf build-mobile-staging && mv build-staging build-mobile-staging",
```
Comment (jowlee) [inline package.json:120]: I don't think you want the `-mobile-` here?

File: src/containers/edit-playlist/desktop/EditPlaylistModal.tsx [hunk 2]
```diff
@@ -59,9 +59,11 @@ const EditPlaylistModal = ({
     setShowDeleteConfirmation(false)
     onClose()
     deletePlaylist(playlistId!)
-    const playlistRoute = playlistPage(handle, title, playlistId!)
-    // If on the playlist page, direct user to feed
-    if (location.pathname === playlistRoute) goToRoute(FEED_PAGE)
+    if (handle && title) {
+      const playlistRoute = playlistPage(handle, title, playlistId!)
+      // If on the playlist page, direct user to feed
+      if (location.pathname === playlistRoute) goToRoute(FEED_PAGE)
```
Comment (jowlee) [inline src/containers/edit-playlist/desktop/EditPlaylistModal.tsx:65]: not `getPathname()`?

**Review by jowlee — APPROVED** [review]
If you go to your account and click on the tabs, it undoes the hash routing. 
In signup there are multiple hashes in the route `/#/signup#password` - it might be fine though -if you reload it just doesn't recognize the route and will redirect to home. - looks like you handled tag search though - has a similar issue.

**Comment by raymondjacobson** [comment]
> If you go to your account and click on the tabs, it undoes the hash routing.
> In signup there are multiple hashes in the route `/#/signup#password` - it might be fine though -if you reload it just doesn't recognize the route and will redirect to home. - looks like you handled tag search though - has a similar issue.

Yeah! I think this is the right behavior actually. Thx for callout, but i double checked this too. The hash routing was only to convince password managers that the URL was changing.

File: package.json (hunk 1 above)
Comment (raymondjacobson) [reply package.json:120]: greattt catch ty

File: src/containers/edit-playlist/desktop/EditPlaylistModal.tsx (hunk 2 above)
Comment (raymondjacobson) [reply src/containers/edit-playlist/desktop/EditPlaylistModal.tsx:65]: dankeshön

**Comment by audius-infra** [comment]
Preview this change https://frontend-demo.audius.co/rj-hash-routing

**Pull request merged**